src/DocLinkChecker/DocLinkChecker/Models/Hyperlink.cs

```diff
@@ -48,6 +48,11 @@ public Hyperlink(string filePath, int line, int col, string url)
                 }
                 else
                 {
+                    // Logic for DocFx RelativePath class is complex <https://github.com/dotnet/docfx/blob/cca05f505e30c5ede36973c4b989fce711f2e8ad/src/Docfx.Common/Path/RelativePath.cs>
+                    // Cannot use `Uri.UnescapeDataString` as it would also decode `%2F` to `/'. DocFX does not do this.
+                    // To avoid writing a complex logic. Just basic replacing of some chars.
+                    Url = Url.Replace("%20", " ").Replace("%7E", "~");
```

Comment:
I wanted to avoid unnecessary complexity. 
All characters outside of the space is very unlikely being used in a markdown link.
The resulting code might be as complex as the one from DocFx and I wanted to avoid this.

Reply:
I would still go for the UrlDecode then, at least it will catch all the %something. And for the %2F, that should just work fie as well looking at the code. The main key point is to find out those edge scenarios.

Reply:
I modified the logic to be a bit closer to DocFx logic, without rebuilding the hole application.
But DocFx has splitted it into multiple classes and is quite complex.

Reply:
Thanks. I see, it's now more complex but seems like it is the right approach. As it's mainly files to validate, I think it's the right approach.